doc/KUKSA.val_data_broker/authorization_alt_d.md

```diff
@@ -0,0 +1,315 @@
+the third party.  This creates several problems and limitations:
+
+* Third-party applications are required to store the resource
+  owner's credentials for future use, typically a password in
+  clear-text.
+
+* Servers are required to support password authentication, despite
+  the security weaknesses inherent in passwords.
+
+* Third-party applications gain overly broad access to the resource
+  owner's protected resources, leaving resource owners without any
+  ability to restrict duration or access to a limited subset of
+  resources.
+
+* Resource owners cannot revoke access to an individual third party
+  without revoking access to all third parties, and must do so by
+  changing the third party's password.
+
+* Compromise of any third-party application results in compromise of
+  the end-user's password and all of the data protected by that
+  password.
+
+OAuth addresses these issues by introducing an authorization layer
+and separating the role of the client from that of the resource
+owner.  In OAuth, the client requests access to resources controlled
+by the resource owner and hosted by the resource server, and is
+issued a different set of credentials than those of the resource
+owner.
+
+Instead of using the resource owner's credentials to access protected
+resources, the client obtains an access token -- a string denoting a
+specific scope, lifetime, and other access attributes.  Access tokens
+are issued to third-party clients by an authorization server with the
+approval of the resource owner.  The client uses the access token to
+access the protected resources hosted by the resource server.
+
+#### Access Token [[RFC 6749](https://www.rfc-editor.org/rfc/rfc6749#section-1.4)]
+
+Access tokens are credentials used to access protected resources.  An
+access token is a string representing an authorization issued to the
+client.  The string is usually opaque to the client.  Tokens
+represent specific scopes and durations of access, granted by the
+resource owner, and enforced by the resource server and authorization
+server.
+
+The token may denote an identifier used to retrieve the authorization
+information or may self-contain the authorization information in a
+verifiable manner (i.e., a token string consisting of some data and a
+signature).  Additional authentication credentials, which are beyond
+the scope of this specification, may be required in order for the
+client to use a token.
+
+The access token provides an abstraction layer, replacing different
+authorization constructs (e.g., username and password) with a single
+token understood by the resource server.  This abstraction enables
+issuing access tokens more restrictive than the authorization grant
+used to obtain them, as well as removing the resource server's need
+to understand a wide range of authentication methods.
+   
+
+
+#### Roles [[RFC 6749](https://www.rfc-editor.org/rfc/rfc6749#section-1.1)]
+
+OAuth defines four roles:
+
```

Comment:
What I like most about adopting OAuth wording, is that we can from an security architecture point of vielw clearly define the scope (pun intended :) ) and responsibilities of KUKSA.val. So to line out my current understanding

 * **resource owner**
 That depends on the use case, it may be the driver, the owner of a vehicle the OEM, etc. On a technical level we wouldn't know

* **resource server**
That is us. KUKSA.val databroker

* **client**
 Everything interacting with our APIs

* **authorization server**
The provider of tokens. Dependent on your application/deployment. It is a "technical component", that is outside the scope of KUKSA.val. Maybe some backend server that is triggered on the fly, if the user gives consent is some UI, maybe some packaging magic, bundling and signing application bundles/updates, etc.

Reply:
Haha. I actually added the exact thing you wrote here (kinda) to the document without seeing your comment here.

PS
I really need to find a workable way to enable Github notifications so I don't miss comments.